qiskit/backends/local/statevector_simulator_cpp.py

```
@@ -0,0 +1,93 @@
+            circuit['compiled_circuit']['operations'].append(
+                {'name': 'snapshot', 'params': [final_state_key]})
+        result = super().run(q_job)._result
+        # Extract final state snapshot and move to 'quantum_state' data field
+        for res in result['result']:
+            snapshots = res['data']['snapshots']
+            if str(final_state_key) in snapshots:
+                final_state_key = str(final_state_key)
+            # Pop off final snapshot added above
+            final_state = snapshots.pop(final_state_key, None)
+            final_state = final_state['quantum_state'][0]
+            # Add final state to results data
+            res['data']['quantum_state'] = final_state
+            # Remove snapshot dict if empty
+            if snapshots == {}:
+                res['data'].pop('snapshots', None)
+        return Result(result, qobj)
+
+    def _validate(self, qobj):
+        """Semantic validations of the qobj which cannot be done via schemas.
+        Some of these may later move to backend schemas.
+
+        1. No shots
+        2. No measurements in the middle
+        """
+        if qobj['config']['shots'] != 1:
+            logger.warning("statevector simulator only supports 1 shot. "
+                           "Setting shots=1.")
+            qobj['config']['shots'] = 1
+        for circuit in qobj['circuits']:
+            if 'shots' in circuit['config'] and circuit['config']['shots'] != 1:
+                logger.warning("statevector simulator only supports 1 shot. "
+                               "Setting shots=1 for circuit %s", circuit['name'])
+                circuit['config']['shots'] = 1
+            for op in circuit['compiled_circuit']['operations']:
+                if op['name'] == 'measure':
+                    raise SimulatorError("In circuit {}: statevector simulator does "
+                                         "not support measure.".format(circuit['name']))
+        return
```

This validation should also raise an error if there are `reset` operations int he circuit.